addons/agent_mirrorpeer_controller.go

```diff
@@ -147,17 +148,29 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if mirrorPeer.Spec.Type == multiclusterv1alpha1.Async && !hasStorageClientRef {
-		clusterFSIDs := make(map[string]string)
-		logger.Info("Fetching clusterFSIDs")
-		err = r.fetchClusterFSIDs(ctx, &mirrorPeer, clusterFSIDs)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				return ctrl.Result{RequeueAfter: 60 * time.Second}, nil
-			}
-			return ctrl.Result{}, fmt.Errorf("an unknown error occurred while fetching the cluster fsids, retrying again: %v", err)
+	clusterFSIDs := make(map[string]string)
+	logger.Info("Fetching clusterFSIDs")
+	err = r.fetchClusterFSIDs(ctx, &mirrorPeer, clusterFSIDs)
```

Comment:
If it is of type `hasStorageClientRef` then will the call the `fetchClusterFSIDs` work or always error out? As it was called within the if prior to this case, wondering if we should retain it as such.

Reply:
Yes, it will work. If `hasStorageClientRef` is true , then these agents would be created on provider clusters. It should always be able to fetch ceph fsids.